sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/_tests_/unit/application-event-date-utils.service.getApplicationEventDate.spec.ts

```diff
@@ -0,0 +1,162 @@
+  let payloadApplication: Pick<
+    Application,
+    "applicationStatus" | "applicationStatusUpdatedOn"
+  >;
+  let payloadDisbursementSchedule: DisbursementScheduleForApplicationEventDate;
+
+  beforeAll(() => {
+    applicationEventDateUtilsService = new ApplicationEventDateUtilsService();
+
+    // Arrange.
+    payloadApplication = {
+      applicationStatus: ApplicationStatus.Enrolment,
+      applicationStatusUpdatedOn: new Date(),
+    };
+    const [feedbackErrorCode] = FULL_TIME_DISBURSEMENT_FEEDBACK_ERRORS;
+    payloadDisbursementSchedule = {
+      updatedAt: addDays(1, new Date()),
+      disbursementDate: addToDateOnlyString(new Date(), -1, "d"),
+      dateSent: addDays(2, new Date()),
+      disbursementFeedbackErrors: [
+        {
+          errorCode: feedbackErrorCode,
+          updatedAt: addDays(-2, new Date()),
+        } as DisbursementFeedbackErrors,
+      ],
+    };
+  });
+
+  it(
+    `Should return payloadApplication.applicationStatusUpdatedOn when application event code ` +
+      `is ${ApplicationEventCode.COER} and application status is ${ApplicationStatus.Enrolment}.`,
+    () => {
+      // Act
+      const applicationEventDate =
+        applicationEventDateUtilsService.getApplicationEventDate(
+          ApplicationEventCode.COER,
+          payloadApplication,
+          payloadDisbursementSchedule,
+        );
+
+      // Assert
+      expect(applicationEventDate).toBe(
+        payloadApplication.applicationStatusUpdatedOn,
+      );
+    },
+  );
+
+  it(
+    `Should return payloadDisbursementSchedule.updatedAt when application event code ` +
+      `is ${ApplicationEventCode.COER} and application status is not ${ApplicationStatus.Enrolment}.`,
+    () => {
+      // Arrange
+      const application = {
+        ...payloadApplication,
+        applicationStatus: ApplicationStatus.Completed,
+      };
+
+      // Act
+      const applicationEventDate =
+        applicationEventDateUtilsService.getApplicationEventDate(
+          ApplicationEventCode.COER,
+          application,
+          payloadDisbursementSchedule,
+        );
+
+      // Assert
+      expect(applicationEventDate).toBe(payloadDisbursementSchedule.updatedAt);
+    },
+  );
+
+  it(`Should return disbursementFeedbackErrors updatedAt when application event code is ${ApplicationEventCode.DISE}.`, () => {
+    // Act
+    const applicationEventDate =
+      applicationEventDateUtilsService.getApplicationEventDate(
+        ApplicationEventCode.DISE,
+        payloadApplication,
+        payloadDisbursementSchedule,
+      );
+
+    // Assert
+    const [{ updatedAt }] =
+      payloadDisbursementSchedule.disbursementFeedbackErrors;
+    expect(applicationEventDate).toBe(updatedAt);
+  });
+
+  it(`Should return payloadDisbursementSchedule.disbursementDate - DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS when application event code is ${ApplicationEventCode.DISR}.`, () => {
```

Comment:
Should `DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS` be evaluated here as the other vars or just a text like "disbursement file generation anticipation days"?